### PR TITLE
mes-3624: Schema for category B+E

### PR DESCRIPTION
**Pull request by MarkPatt89**
# Description and relevant Jira numbers
Introduction of a new schema for category B+E vehicles which includes:

- adding`businessDetails`
- adding `reverseLeft` as the only manoeuvre
- removing `instructorReg`
- refactor of `vehicleChecks` for multiple `showMe/tellMe` questions
## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [x] Reviewers selected in Github
- [x] PR link added to JIRA

### CODE REVIEW

File: mes-test-schema/categories/BE/index.json [hunk 1]
```diff
@@ -0,0 +1,1776 @@
...
+          "$ref": "#/definitions/gearboxCategory"
+        },
+        "schoolCar": {
+          "description": "Indicates whether the vehicle belongs to a driving school",
+          "type": "boolean"
+        },
+        "dualControls": {
+          "description": "Indicates whether or not the vehicle has dual controls fitted",
+          "type": "boolean"
+        }
```
Comment (rossfauldsbjss) [inline mes-test-schema/categories/BE/index.json]: on cat c i removed these, worth checking if cat b+e needs them

Comment (MarkPatt89) [reply mes-test-schema/categories/BE/index.json]: good spot, thanks Ross

File: mes-test-schema/categories/BE/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,956 @@
...
+  ETA?: ETA;
+  faultSummary?: FaultSummary;
+  eyesightTest?: EyesightTest;
+}
+/**
+ * Details of the Show Me and Tell Me questions asked during the test
+ */
+export interface VehicleChecks {
+  tellMeQuestions?: {
+    code?: QuestionCode;
```
Comment (Cavalier07) [inline mes-test-schema/categories/BE/index.d.ts]: do we need to order these? Which question was done first , second and third

File: mes-test-schema/categories/BE/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,956 @@
...
+  interpreter?: boolean;
+  /**
+   * Indicates whether another individual was present during the test
+   */
+  other?: boolean;
+}
+/**
+ * Details about the vehicle to be used for the test
+ */
+export interface VehicleDetails {
```
Comment (Cavalier07) [inline mes-test-schema/categories/BE/index.d.ts]: I would assume we need the length/width of the car in here or in the journal data section

File: mes-test-schema/categories/BE/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,956 @@
...
+   */
+  provisionalLicenceProvided: boolean;
+  /**
+   * The PCN issued to the candidate
+   */
+  passCertificateNumber: string;
+  /**
+   * Indicate presence of code 78 (automatic) on candidates license
+   */
+  code78Present?: boolean;
```
Comment (Cavalier07) [inline mes-test-schema/categories/BE/index.d.ts]: this feels like it should be mandatory

File: mes-test-schema/categories/BE/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,956 @@
...
+export interface Manoeuvres {
+  reverseLeft?: {
+    selected?: ManoeuvreIndicator;
+    controlFault?: ManoeuvreOutcome;
+    observationFault?: ManoeuvreOutcome;
+    controlFaultComments?: FaultComments;
+    observationFaultComments?: FaultComments;
+  };
+}
+export interface UncoupleRecouple {
```
Comment (Cavalier07) [inline mes-test-schema/categories/BE/index.d.ts]: Could see this being inside of Manoeuvres

Comment (MarkPatt89) [reply mes-test-schema/categories/BE/index.d.ts]: According to the Miro and confluence docs, the only manoeuvre required for B+E is `reverseLeft`.

File: mes-test-schema/categories/BE/index.d.ts (hunk 4 above)
Comment (MarkPatt89) [reply mes-test-schema/categories/BE/index.d.ts]: Spoke with Oli, yes it should be.

File: mes-test-schema/categories/BE/index.d.ts (hunk 3 above)
Comment (MarkPatt89) [reply mes-test-schema/categories/BE/index.d.ts]: I missed that. I think it more logical to contain it within `VehicleDetails`

File: mes-test-schema/categories/BE/index.d.ts (hunk 5 above)
Comment (Cavalier07) [reply mes-test-schema/categories/BE/index.d.ts]: Is UncoupleRecouple a manoeuvre? or should we class it as one as it uses the same structures underneath

**Pull request merged**